File: adminpanel/NexoraAdminPanel/src/NexoraAcademy.AdminBff/Auth/Roles.cs
namespace NexoraAcademy.AdminBff.Auth;

public static class Roles
{
    public const string AdminOnly = "ADMIN,SYSTEM_ADMIN";
    public const string ContentManager = "ADMIN,SYSTEM_ADMIN,CONTENT_MANAGER";
    public const string SalesCrm = "ADMIN,SYSTEM_ADMIN,SALES_CRM";
}

File: frontend/main-site/src/auth/roles.ts
export const ROLE_GROUPS = {
  adminOnly: ["ADMIN", "SYSTEM_ADMIN"],
  contentManager: ["ADMIN", "SYSTEM_ADMIN", "CONTENT_MANAGER"],
  salesCrm: ["ADMIN", "SYSTEM_ADMIN", "SALES_CRM"],
  panelAccess: ["ADMIN", "SYSTEM_ADMIN", "CONTENT_MANAGER", "SALES_CRM"],
} as const;

export type RoleGroup = keyof typeof ROLE_GROUPS;

export function hasRole(userRole: string | undefined, group: RoleGroup): boolean {
  if (!userRole) return false;
  return (ROLE_GROUPS[group] as readonly string[]).includes(userRole);
}

export const ROLE_LABELS: Record<string, string> = {
  SALES_CRM: "Satış/CRM",
  CONTENT_MANAGER: "Kontent Meneceri",
  ADMIN: "Admin",
  SYSTEM_ADMIN: "Sistem Admini",
};
